src/modules/0.0.22/aws_codedeploy/sql/after_install.sql

```diff
@@ -0,0 +1,140 @@
+ -- prevents update on revision table
+CREATE
+OR REPLACE FUNCTION block_codedeploy_revision_update_function () RETURNS TRIGGER AS $block_codedeploy_revision_update_function$
+    BEGIN
+        RAISE EXCEPTION 'Revision cannot be modified'
+        USING detail = 'Revisions can only be added';
+        RETURN OLD;
+    END;
+$block_codedeploy_revision_update_function$ LANGUAGE plpgsql;
+
+CREATE TRIGGER
+  block_codedeploy_revision_update_trigger BEFORE
+UPDATE
+  ON codedeploy_revision
+EXECUTE
+  FUNCTION block_codedeploy_revision_update_function ();
+
+-- prevents deletion on revision table
+CREATE
+OR REPLACE FUNCTION block_codedeploy_revision_delete_function () RETURNS TRIGGER AS $block_codedeploy_revision_delete_function$
+    BEGIN
+        RAISE EXCEPTION 'Revision cannot be deleted'
+        USING detail = 'Revisions can only be added';
+        RETURN OLD;
+    END;
+$block_codedeploy_revision_delete_function$ LANGUAGE plpgsql;
+
+CREATE TRIGGER
+  block_codedeploy_revision_delete_trigger BEFORE DELETE ON codedeploy_revision
+EXECUTE
+  FUNCTION block_codedeploy_revision_delete_function ();
```

Comment:
it cannot delete rows created by the user, it only allows deletion on cascade. So the rows would only be deleted when the application is. I had to do that way because i do not have a simple way to identify an unique revision. They do not have any ID, any arn, name... you can have 2 revisions totally identical, the only thing that could be different is the creation date.

Reply:
But this would put the database into a broken state where apply and sync don't function anymore, correct?

Reply:
not sure if i follow you... people can only add new revisions if they are associated with an application. I also do not reconcile on updates, just on inserts. In the application i only reconcile when a new revision is added to the cloud, i check the length of the revisions array. However i'm happy to hear about suggestions on how to improve it , or on new tests i could add to improve coverage..

Reply:
mm are you worried about the atomicity of these rules? if so, do you know a better way to block deletes but allow deletes on cascade?